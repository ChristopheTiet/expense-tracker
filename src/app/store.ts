import { configureStore } from "@reduxjs/toolkit";
import expensesReducer from "../ducks/expenses-slice";
import filterReducer from "../ducks/filter-slice";

export const store = configureStore({
  reducer: {
    expenses: expensesReducer,
    filter: filterReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
