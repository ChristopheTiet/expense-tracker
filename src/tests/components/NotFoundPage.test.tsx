import renderer from "react-test-renderer";
import NotFoundPage from "../../components/NotFoundPage";
import { BrowserRouter } from "react-router-dom";
import { store } from "../../app/store";
import { Provider } from "react-redux";

test("should render Header correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <NotFoundPage />
        </Provider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
