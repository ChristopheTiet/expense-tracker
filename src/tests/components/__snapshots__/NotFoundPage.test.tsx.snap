// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<div>
  404 - 
  <a
    href="/"
    onClick={[Function]}
  >
    Go home
  </a>
</div>
`;
